ons-spark/spark-analysis/Cramer_v.md

```diff
@@ -0,0 +1,439 @@
+## Calculating Cramér's V from a Spark DataFrame
+
+### What is Cramér's V?
+Cramér's V is a statistical measure of an association between two nominal variables, giving a value between 0 and 1 inclusive. Here 0 would indicate no association and 1 indicates a strong association between the two variables. It is based on Pearson's chi-square statistics.
+
+We calculate Cramér's V as follows:
+
+$$ \text{Cramer's V} = \sqrt{\dfrac{\dfrac{\chi^2}{n}}{\min (c-1,r-1)}}, $$ 
+where:
+- $\chi^2$ is the Chi-squared statistic,
+- $n$ is the number of samples,
+- $r$ is the number of rows,
+- $c$ is the number of columns.
+
+In some literature you may see the Phi coefficient used ($\phi$), where $\phi^2 = \chi^2/n$.
+
+### Cramér's V in Spark:
+Although there is not an in built method for calculating this statistic in base python, is it reasonably straightforward using `numpy` and `scipy.stats` packages. An example of this can be found [online here](https://www.statology.org/cramers-v-in-python/).
+A similar example for R is linked [here](https://www.statology.org/cramers-v-in-r/).
+
+To calculate the Cramér V statistic, we will need to first calculate the $\chi^2$ statistic. In python we will utilise `scipy.stats.chi2_contingency` and `chisq.test` in R. Both these functions will take a matrix like input of a contingency table / pair-wise frequency table. Both Pyspark and SparklyR have inbuilt functions which can produce these tables (`crosstab`/`sdf_crosstab`) as we will see shortly.
+
+Due to Pyspark and SparklyR's differences to classical python and R, we need to consider how we can calculate Cramér's V when using Spark DataFrames.
+First we will import the needed packages, start a spark session and load the rescue data. 
+````{tabs}
+```{code-tab} py
+import yaml
+import numpy as np
+from pyspark.sql import SparkSession, functions as F
+import scipy.stats as stats
+
+
+spark = (SparkSession.builder.master("local[2]")
+         .appName("cramer-v")
+         .getOrCreate())
+
+
+with open("../../../config.yaml") as f:
```

Comment:
So here we go 3 level above , but in the relevant R code we just use ons-spark/config.yaml .  ( ons-spark.config.yaml worked for me )

Reply:
Changing to ons-spark.config.yaml did not work in the raw notebook. Going to leave as is, there is an open issue which will look at file paths in the future